### PR TITLE
rip out metatags from action methods and metaprogram our way into awesom...

### PR DESCRIPTION
**Pull request by crdunwel**
...eness

### CODE REVIEW

**Comment by handlers** [comment]
This is neat! A couple of things:

* We probably want to make sure that we would be comfortable [maintaining this gem](https://github.com/lowjoel/before_render/tree/master/lib/before_render), which appears to be forked by some enterprising maintainer every so often.
* This solution relies on building up some institutional knowledge around how to set meta tags, so we'd need to update the [wiki](https://github.com/sunlightlabs/opencongress/wiki/Meta-Tags). I personally prefer the explicitness that comes with setting the meta tags in the controller, but not so much that I'd reject this solution.

Any thoughts, @Cspeisman?

**Comment by handlers** [comment]
Er, that should reads "in the controller action itself"

**Comment by crdunwel** [comment]
- The gem should be fine for rails 4 ... so the foreseeable future. Seems the gem has persisted in one form or another since rails 2. In the event that we lose support for the gem then we could alternatively override the "render" method in application_controller and run the metatag generation methods before calling super. This is what I did before using the gem and achieves the same end result as the current solution (I just like the current solution because it's a little cleaner and extendable)
- Institutional knowledge isn't a bad thing - I'd say it's actually pretty valuable. I don't think the system is too complicated that one couldn't immediately see what's going on either. 

I'm not entirely opposed to putting the metatag methods in the action themselves. I just think it's cluttering the action methods. Also, what if you wanted to get metatags for an action without the accompanying HTML? Separation of concerns is a sound design principle. Another thing to consider is what if you want the metatags to use data generated in the action itself? In the current setup you'd have to put the metatags below the other action code. What if you were rendering a different template than the rails naming convention would otherwise dictate? You'd have metatag code below the other action code but before the render method. That can get pretty messy with metatag code in different places in different actions. In my solution, however, the metatags would always be set right before calling render so all other context variables are assured to be present.

tl;dr

- Cleaner
- Separation of Concerns
- Slightly DRYer

**Comment by Cspeisman** [comment]
@handlers @crdunwel I'm okay with the solution.  Gonna merge.
